python/segyio/trace.py

```diff
@@ -842,9 +843,26 @@ def __setitem__(self, i, val):
 
         >>> f.text[1] = g.text[0]
 
+        If i is slice [i_1:i_n] and val=[val_1, .. val_n],
+        text[k] = val[k] for k in 1..n
+
+        >>> f.text[1:3] = ["new_header1", "new_header_2"]
+
+        If val is instance of Text, text[i] = val[0]
+
+        >>> f.text[1] = g.text
+        >>> assert f.text[1] == g.text[0]
+
+        If i is slice [i_1:i_n] and val is [Text_1, ... Text_n],
+        text[k] = val[k][0] for k in 1..n
+
+        >>> f.text[1:3] = [g1.text, g2.text]
+        >>> assert f.text[1] == g1.text[0]
+        >>> assert f.text[2] == g2.text[0]
```

Comment:
Please review the numpy docstring format - this entire section will probably not render well.

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html

You can build the documentation locally by building with `cmake -DBUILD_DOC=ON` and running `make doc`